src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -1345,9 +1345,49 @@ void Terminal::_updateUrlDetection()
     }
 }
 
-// Interns URegularExpression instances so that they can be reused. This method is thread-safe.
-static ICU::unique_uregex internURegularExpression(const std::wstring_view& pattern)
+struct URegularExpressionInterner
 {
+    // Interns (caches) URegularExpression instances so that they can be reused. This method is thread-safe.
+    // uregex_open is not terribly expensive at ~10us/op, but it's also much more expensive than uregex_clone
+    // at ~400ns/op and would effectively double the time it takes to scan the viewport for patterns.
+    //
+    // An alternative approach would be to not make this method thread-safe and give each
+    // Terminal instance its own cache. I'm not sure which approach would've been better.
+    ICU::unique_uregex Intern(const std::wstring_view& pattern)

```

Comment:
FYI suppress whitespace changes. The conversion to a class barely changed anything.